Makefile.am

```diff
@@ -215,6 +215,15 @@ _usnic_files = \
 
 if HAVE_VERBS
 _usnic_files += prov/usnic/src/usdf_fake_ibv.c
+
+# If the verbs or usnic providers are being built as a DL, then we
+# need to link libibverbs into the usnic provider
+if HAVE_VERBS_DL
+_usnic_libibverbs = -libverbs
+endif
+if HAVE_USNIC_DL
+_usnic_libibverbs = -libverbs
+endif
 endif
```

Comment:
A clarification: the usnic provider only links against libibverbs if the verbs provider is being built.  As such:

- we know that libibverbs is present (and don't have to add additional configure.m4 checks to ensure that it is there)
- we know that we only need to add -libverbs if the usnic provider will not be in the same file as the verbs provider

> On May 5, 2015, at 12:44 PM, Patrick McCormick <notifications@github.com> wrote:
> 
> In Makefile.am:
> 
> > @@ -215,6 +215,15 @@ _usnic_files = \
> >  
> >  if HAVE_VERBS
> >  _usnic_files += prov/usnic/src/usdf_fake_ibv.c
> > +
> > +# If the verbs or usnic providers are being built as a DL, then we
> > +# need to link libibverbs into the usnic provider
> > +if HAVE_VERBS_DL
> > +_usnic_libibverbs = -libverbs
> > +endif
> > +if HAVE_USNIC_DL
> > +_usnic_libibverbs = -libverbs
> > +endif
> >  endif
> 
> I still don't think we really need either. usnic provider now needs to link against libibverbs, just like the verbs provider. So have them both add this in their configure.m4, just like how the verbs provider does it.
> 
> The linker will just ignore multiple requests to link against the same library and just link once.
> 
> —
> Reply to this email directly or view it on GitHub.
> 


-- 
Jeff Squyres
jsquyres@cisco.com
For corporate legal information go to: http://www.cisco.com/web/about/doing_business/legal/cri/